### ISSUE TITLE
extend to support sockets via arbitrary file descriptors?

### ISSUE THREAD

**Issue by subes** [zrlio/jaio#4]
Would be interesting to experiment with this as an alterantive to EPOLL as described here: https://blog.cloudflare.com/io_submit-the-epoll-alternative-youve-never-heard-about/

Though from what I see currently the API only supports actual "File" instances, not arbitrary file descriptors (which we can also extract from SocketChannels via reflection).

**Comment by subes** [comment]
I guess we would only require an alternative constuctor for this class that allows to put in the fd pointer directly: https://github.com/zrlio/jaio/blob/master/src/main/java/com/ibm/jaio/File.java
The path is not used anyway?

**Comment by PepperJo** [comment]
Yes you are correct. The path member is currently not used. So it should be an easy change. Feel free to create a PR which adds a constructor and removes the path member.
IMO the best current solution for Linux is to use `io_uring`: https://github.com/axboe/liburing/wiki/io_uring-and-networking-in-2023
Maybe someone already created a wrapper for Java?

**Comment by subes** [comment]
Thanks for the quick answer and confirming my idea, will try that.

There is this netty integration: https://github.com/netty/netty-incubator-transport-io_uring
Dunno how easy it will be to use it without netty.

I will create a PR when I tackle alternatives to nio selector. Might take a few more weeks until I get to experimenting with jaio.
